### PR TITLE
jbpm persistence and transactions example update

### PR DESCRIPTION
**Pull request by jsvitak**
Let me know if this change is ok and if possible merge the PR. There are more places where StatefulKnowledgeSession is used in jbpm-docs, is this ok? Because I think that StatefulKnowledgeSession has been deprecated in favor of KieSession.

### CODE REVIEW

**Comment by psiroky** [comment]
@krisv, @mrietveld or @mswiderski could you please review this?

File: kie-docs/jbpm-docs/src/main/docbook/en-US/Persistence/Transactions.xml [hunk 1]
```diff
@@ -22,22 +22,34 @@
...
-EntityManagerFactory emf = Persistence.createEntityManagerFactory( "org.jbpm.persistence.jpa" );
-Environment env = KnowledgeBaseFactory.newEnvironment();
-env.set( EnvironmentName.ENTITY_MANAGER_FACTORY, emf );
-env.set( EnvironmentName.TRANSACTION_MANAGER, TransactionManagerServices.getTransactionManager() );
+// create the entity manager factory
+EntityManagerFactory emf = EntityManagerFactoryManager.get().getOrCreate("org.jbpm.persistence.jpa");
+TransactionManager tm = TransactionManagerServices.getTransactionManager();
+Environment env = EnvironmentFactory.newEnvironment();
+env.set(EnvironmentName.ENTITY_MANAGER_FACTORY, emf);
+env.set(EnvironmentName.TRANSACTION_MANAGER, tm);
```
Comment (mswiderski) [inline kie-docs/jbpm-docs/src/main/docbook/en-US/Persistence/Transactions.xml]: you don't need to setup environment here

File: kie-docs/jbpm-docs/src/main/docbook/en-US/Persistence/Transactions.xml [hunk 2]
```diff
@@ -22,22 +22,34 @@
...
+
+// setup the runtime environment
+RuntimeEnvironment environment = RuntimeEnvironmentBuilder.Factory.get()
+.newDefaultBuilder()
+.addAsset(ResourceFactory.newClassPathResource("MyProcessDefinition.bpmn2"), ResourceType.BPMN2)
+    .addEnvironmentEntry(EnvironmentName.TRANSACTION_MANAGER, tm)
+    .addEnvironmentEntry(EnvironmentName.PERSISTENCE_CONTEXT_MANAGER, new JpaProcessPersistenceContextManager(env))
+    .addEnvironmentEntry(EnvironmentName.TASK_PERSISTENCE_CONTEXT_MANAGER, new JPATaskPersistenceContextManager(env))
```
Comment (mswiderski) [inline kie-docs/jbpm-docs/src/main/docbook/en-US/Persistence/Transactions.xml]: no need to define manually persistence context managers, they will be setup automatically by the engine

**Comment by mswiderski** [comment]
besides these two comments all is good from my point of view

**Comment by jsvitak** [comment]
Hello Maciej, I have removed the persistence context managers initialization from the example. Can you check now? Thanks.

**Comment by mswiderski** [comment]
looks good now

**Pull request merged**